### ISSUE TITLE
Routed paths must always start with /

### ISSUE THREAD

**Issue by vitalik** [tiangolo/fastapi#414]
Hi

Let's say I have the following API:

```
GET /accounts  - list of accounts
POST /accounts  - create account
GET /accounts/{id} - get account by id
DELETE  /accounts/{id} - delete account
```

then I would make a module accounts with APIRouter

in main.py

`app.include_router(accounts.router, prefix="/accounts")`

and in accounts.py

router = APIRouter()

```
@router.get("")
def list_accounts():
    ...

@router.post("")
def list_accounts():
    ...

@router.get("/{id}")
def get_account(id:str):
    ...
```

but I'm getting an error "Routed paths must always start with /"

if I add to router "/" - then my resources will look like  `GET /accounts/` ,  `POST /accounts/`  (but I do not want slash at the end...)

Basically the idea is to put 5-10 methods inside module and all should start with /accounts

how should i solve this ?

**Comment by euri10** [comment]
You can add a prefix to your router

**Comment by vitalik** [comment]
@euri10 yeah, I did - my prefix="/accounts"  - but i cannot make api resource  /accounts (without ending slash) now with routers

**Comment by dmontagu** [comment]
I think you'd have to do:

```python
app.include_router(accounts.router, prefix="")  # can drop the prefix kwarg
@router.get("/accounts")
def list_accounts():
    ...

@router.post("/accounts")
def list_accounts():
    ...

@router.get("/accounts/{id}")
def get_account(id:str):
    ...
```

You can mount multiple routers with the same prefix, so this won't cause a problem if you want to use a different router for each resource.

You'd have to make sure you didn't create conflicts by naming routers the same thing, but that's always a risk anyway.

**Comment by vitalik** [comment]
but what's the technical reason not to allow empty router if parent-include have prefix ?

**Comment by jaddison** [comment]
It does seem sort of awkward that using `prefix` prevents 'trailing slash' naked endpoints, as OP mentioned.

Good: `/cats`
Bad: `/cats/`

(subjectively speaking)

**Comment by tiangolo** [comment]
Thank you, everyone, for the discussion here!

Thanks @vitalik for the PR! More comments on the PR itself: https://github.com/tiangolo/fastapi/pull/415#issuecomment-524591290

You can now use this in FastAPI `0.36.0` with an empty path (`""`). :tada: :rocket: 

---

I'll close this now, but feel free to add more comments or create new issues.
